sympy/solvers/solvers.py

```diff
@@ -2750,6 +2750,9 @@ def nsolve(*args, **kwargs):
         f = args[0]
         fargs = None
         x0 = args[1]
+        for func0 in f:
+            if len(func0.free_symbols)>1:
```

Comment:
This test isn't correct. If f is an iterable and only two arguments are given, the function should error, regardless of how many free_symbols there are (three arguments need to be given in this case).